src/sentry/snuba/sessions.py

```diff
@@ -0,0 +1,167 @@
+        selected_columns=["project_id", "release"],
+        groupby=["release", "project_id"],
+        orderby=[["divide", ["%s_crashed" % scope, scope]]],
+        start=yesterday,
+        offset=offset,
+        limit=limit,
+        conditions=conditions,
+        filter_keys=filter_keys,
+    )["data"]:
+        rv.append((x["project_id"], x["release"]))
+
+    return rv
+
+
+def _make_stats(start, rollup, buckets):
+    rv = []
+    start = int(to_timestamp(start) // rollup) * rollup
+    for x in range(buckets):
+        rv.append([start, 0])
+        start += rollup
+    return rv
+
+
+def get_release_health_data_overview(project_releases, environments=None, stats_period=None):
+    """Checks quickly for which of the given project releases we have
+    health data available.  The argument is a tuple of `(project_id, release_name)`
+    tuples.  The return value is a set of all the project releases that have health
+    data.
+    """
+
+    def _nan_as_none(val):
+        return None if val != val else val
+
+    yesterday = datetime.now(pytz.utc) - timedelta(days=1)
+    conditions, filter_keys = _get_conditions_and_filter_keys(project_releases, environments)
+
+    if stats_period == "24h":
+        stats_rollup = 3600
+        stats_start = yesterday
+        stats_buckets = 24
+    elif stats_period == "14d":
+        stats_rollup = 86400
+        stats_start = datetime.now(pytz.utc) - timedelta(days=14)
+        stats_buckets = 14
+    elif not stats_period:
+        stats_rollup = None
+        stats_start = None
+    else:
+        raise TypeError("Invalid stats period")
+
+    total_users_24h = {}
+    for x in raw_query(
+        dataset=Dataset.Sessions,
+        selected_columns=["release", "users"],
+        groupby=["release", "project_id"],
+        start=yesterday,
+        conditions=conditions,
+        filter_keys=filter_keys,
+    )["data"]:
+        total_users_24h[x["project_id"]] = x["users"]
+
+    rv = {}
+    for x in raw_query(
+        dataset=Dataset.Sessions,
+        selected_columns=[
+            "release",
+            "project_id",
+            "duration_quantiles",
+            "users",
+            "sessions",
+            "sessions_errored",
+            "sessions_crashed",
+            "users_crashed",
+        ],
+        groupby=["release", "project_id"],
+        start=yesterday,
+        conditions=conditions,
+        filter_keys=filter_keys,
+    )["data"]:
+        total_users = total_users_24h.get(x["project_id"])
+        rp = {
+            "duration_p50": _nan_as_none(x["duration_quantiles"][0]),
+            "duration_p90": _nan_as_none(x["duration_quantiles"][1]),
+            "crash_free_users": (
+                100 - x["users_crashed"] / float(x["users"]) * 100 if x["users"] else None
+            ),
+            "crash_free_sessions": (
+                100 - x["sessions_crashed"] / float(x["sessions"]) * 100 if x["sessions"] else None
+            ),
```

Comment:
You should be able to use snuba's aggregations to do these calculations in clickhouse.

Reply:
Is there a benefit to doing that?